actionview/lib/action_view/template/resolver.rb

```diff
@@ -17,9 +17,11 @@ class PathParser # :nodoc:
       ParsedPath = Struct.new(:path, :details)
 
       def build_path_regex
-        handlers = Template::Handlers.extensions.map { |x| Regexp.escape(x) }.join("|")
-        formats = Template::Types.symbols.map { |x| Regexp.escape(x) }.join("|")
-        locales = "[a-z]{2}(?:[-_][A-Z]{2})?"
+        handlers = Regexp.union(Template::Handlers.extensions.map(&:to_s))
+        formats = Regexp.union(Template::Types.symbols.map(&:to_s))
+        available_locales = I18n.available_locales.map(&:to_s)
+        regular_locales = [/[a-z]{2}(?:[-_][A-Z]{2})?/]
+        locales = Regexp.union(available_locales + regular_locales)
```

Comment:
```suggestion
        locales = Regexp.union(available_locales, regular_locales)
```

I think you can make `available_locales` just the regex and avoid the array allocation here

Reply:
It's a good idea, but I don't think it will work because `available_locales` is an array. `Regexp.union` will only accept either a single argument array, or many arguments of patterns.

Reply:
Yep I tried that approach before giving the suggestion, but `Regexp.union` only accepts `array` as argument and not many arguments pattern.